File: level1/re-id/solution-chris.scala
def solution(n: Int): String = {
  def isPrime(x: Int): Boolean = {
    val limit = Math.sqrt(x).toInt
    !(2 to limit).exists(x % _ == 0)
  }

  def generatePrimes(target: Int, current: Int = 2, soFar: String = ""): String = {
    if (target > soFar.length){
      val updated = if(isPrime(current)){ soFar + current } else { soFar }
      generatePrimes(target, current + 1, updated)
    } else {
      soFar
    }
  }

  generatePrimes(n + 5).slice(n, n + 5)
}

File: level1/re-id/solution-chris.py
from math import sqrt


def isPrime(x):
    current = 2
    limit = int(sqrt(x))

    while limit >= current:
        if x % current == 0:
            return False
        current += 1

    return True


def solution(n):
    target = n + 5
    so_far = ""
    current = 2

    while target > len(so_far):
        if isPrime(current):
            so_far = so_far + str(current)
        current += 1

    return so_far[n:target]



